geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/legacy/id/RecordQueryableIndex.scala

```diff
@@ -71,6 +74,20 @@ trait RecordQueryableIndex extends AccumuloFeatureIndex
       } else if (hints.isDensityQuery) {
         val iter = KryoLazyDensityIterator.configure(sft, this, filter.secondary, hints, dupes)
         (Seq(iter), KryoLazyDensityIterator.kvsToFeatures(), None)
+      } else if (hints.isArrowQuery) {
```

Comment:
As a discussion point, should the 'legacy' iterators get updates to handle new capabilities like this?

Reply:
in this case we have customers that can't update their data to the latest format but want arrow functionality...